src/vs/workbench/parts/terminal/electron-browser/terminalActions.ts

```diff
@@ -657,22 +657,39 @@ export class RunActiveFileInTerminalAction extends Action {
 		}
 		let uriPath: string = uri.fsPath;
 		const hasSpace = uriPath.indexOf(' ') !== -1;
-		if (hasSpace && isWindows) {
-			uriPath = '"' + uriPath + '"';
-		} else if (!isWindows) {
-			if (uriPath.indexOf('\\') !== 0) {
-				uriPath = uriPath.replace(/\\/g, '\\\\');
+		if (isWindows) {
+			const exe = this.terminalService.getActiveInstance().shellLaunchConfig.executable;
+			const osVersion = (/(\d+)\.(\d+)\.(\d+)/g).exec(os.release());
```

Comment:
We could expose a `ITerminalInstance.preparePathForTerminal` method which takes a path and quotes it/runs it through `wslpath`, then this stuff is contained inside `TerminalInstance` with similar code (the `os.release()` parsing can be shared).